domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/domain/attendance/AttendanceWorkflowImplementation.kt

```diff
@@ -0,0 +1,76 @@
+package com.krzykrucz.elesson.currentlesson.domain.attendance
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.ClassRegistry
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.LessonIdentifier
+import java.time.LocalDateTime
+
+fun noteAbsence(
+        isLessonStarted: IsLessonStarted,
+        isInRegistry: IsInRegistry,
+        areAllStudentsChecked: AreAllStudentsChecked
+): NoteAbsence = { lessonId, uncheckedStudent, attendanceList, classRegistry ->
+    noteStudent(isLessonStarted, lessonId, isInRegistry, uncheckedStudent, classRegistry, attendanceList, areAllStudentsChecked, addAbsentStudent())
+}
+
+fun notePresence(
+        isLessonStarted: IsLessonStarted,
```

Comment:
this check is unnecessary, because when we take attendanceList as an input, we are sure that lesson is started, because empty AttendanceList is created from StartedLesson (maybe StartedLesson should be taken as an input...)

Reply:
Ok